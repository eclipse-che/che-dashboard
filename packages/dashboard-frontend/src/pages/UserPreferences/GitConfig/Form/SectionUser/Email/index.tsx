/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { FormGroup, TextInput, ValidatedOptions } from '@patternfly/react-core';
import { ExclamationCircleIcon } from '@patternfly/react-icons';
import React from 'react';

import { InputGroupExtended } from '@/components/InputGroupExtended';

const ERROR_REQUIRED_VALUE = 'A value is required.';
const MAX_LENGTH = 128;
const ERROR_MAX_LENGTH = `The value is too long. The maximum length is ${MAX_LENGTH} characters.`;
const REGEX = /^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}$/;
const ERROR_INVALID_EMAIL = 'The value is not a valid email address.';

export type Props = {
  isLoading: boolean;
  value: string;
  onChange: (value: string, isValid: boolean) => void;
};
export type State = {
  validated: ValidatedOptions | undefined;
  value: string | undefined;
};

export class GitConfigUserEmail extends React.PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      validated: ValidatedOptions.default,
      value: props.value,
    };
  }

  public componentDidUpdate(_prevProps: Readonly<Props>, prevState: Readonly<State>): void {
    if (prevState.value === this.state.value && this.props.value !== this.state.value) {
      // reset the initial value
      this.setState({
        value: this.props.value,
        validated: ValidatedOptions.default,
      });
    }
  }

  private handleChange(value: string): void {
    const validate = this.validate(value);
    const isValid = validate === ValidatedOptions.success;

    this.setState({
      value,
      validated: validate,
    });
    this.props.onChange(value, isValid);
  }

  private validate(value: string): ValidatedOptions {
    if (value.length === 0) {
      return ValidatedOptions.error;
    }
    if (value.length > MAX_LENGTH) {
      return ValidatedOptions.error;
    }
    if (!REGEX.test(value)) {
      return ValidatedOptions.error;
    }
    return ValidatedOptions.success;
  }

  public render(): React.ReactElement {
    const { isLoading } = this.props;
    const { value = '', validated } = this.state;

    let errorMessage: string;
    if (value.length === 0) {
      errorMessage = ERROR_REQUIRED_VALUE;
    } else if (value.length > MAX_LENGTH) {
      errorMessage = ERROR_MAX_LENGTH;
    } else if (!REGEX.test(value)) {
      errorMessage = ERROR_INVALID_EMAIL;
    } else {
      errorMessage = '';
    }

    const fieldId = 'gitconfig-user-email';

    return (
      <FormGroup
        label="email"
        fieldId={fieldId}
        isRequired
        helperTextInvalid={errorMessage}
        helperTextIcon={<ExclamationCircleIcon />}
        validated={validated}
      >
        <InputGroupExtended
          isLoading={isLoading}
          readonly={false}
          required={true}
          validated={validated}
          value={value}
          onRemove={undefined}
        >
          <TextInput
            id={fieldId}
            isDisabled={isLoading}
            validated={validated}
            value={value}
            onChange={value => this.handleChange(value)}
          />
        </InputGroupExtended>
      </FormGroup>
    );
  }
}
