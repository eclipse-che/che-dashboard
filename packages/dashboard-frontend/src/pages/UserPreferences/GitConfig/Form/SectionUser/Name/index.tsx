/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { FormGroup, TextInput, ValidatedOptions } from '@patternfly/react-core';
import { ExclamationCircleIcon } from '@patternfly/react-icons';
import React from 'react';

import { InputGroupExtended } from '@/components/InputGroupExtended';

const ERROR_REQUIRED_VALUE = 'A value is required.';
const MAX_LENGTH = 128;
const ERROR_MAX_LENGTH = `The value is too long. The maximum length is ${MAX_LENGTH} characters.`;

export type Props = {
  isLoading: boolean;
  value: string;
  onChange: (value: string, isValid: boolean) => void;
};
export type State = {
  validated: ValidatedOptions | undefined;
  value: string | undefined;
};

export class GitConfigUserName extends React.PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      validated: ValidatedOptions.default,
      value: props.value,
    };
  }

  public componentDidUpdate(_prevProps: Readonly<Props>, prevState: Readonly<State>): void {
    if (prevState.value === this.state.value && this.props.value !== this.state.value) {
      // reset the initial value
      this.setState({
        value: this.props.value,
        validated: ValidatedOptions.default,
      });
    }
  }

  private handleChange(value: string): void {
    const validated = this.validate(value);
    const isValid = validated === ValidatedOptions.success;

    this.setState({
      value,
      validated,
    });
    this.props.onChange(value, isValid);
  }

  private validate(value: string): ValidatedOptions {
    if (value.length === 0) {
      return ValidatedOptions.error;
    }
    if (value.length > MAX_LENGTH) {
      return ValidatedOptions.error;
    }

    return ValidatedOptions.success;
  }

  public render(): React.ReactElement {
    const { isLoading } = this.props;
    const { value = '', validated } = this.state;

    let errorMessage: string;
    if (value.length === 0) {
      errorMessage = ERROR_REQUIRED_VALUE;
    } else if (value.length > MAX_LENGTH) {
      errorMessage = ERROR_MAX_LENGTH;
    } else {
      errorMessage = '';
    }

    const fieldId = 'gitconfig-user-name';

    return (
      <FormGroup
        label="name"
        fieldId={fieldId}
        isRequired
        helperTextInvalid={errorMessage}
        helperTextIcon={<ExclamationCircleIcon />}
        validated={validated}
      >
        <InputGroupExtended
          isLoading={isLoading}
          readonly={false}
          required={true}
          validated={validated}
          value={value}
          onRemove={undefined}
        >
          <TextInput
            id={fieldId}
            isDisabled={isLoading}
            validated={validated}
            value={value}
            onChange={value => this.handleChange(value)}
          />
        </InputGroupExtended>
      </FormGroup>
    );
  }
}
