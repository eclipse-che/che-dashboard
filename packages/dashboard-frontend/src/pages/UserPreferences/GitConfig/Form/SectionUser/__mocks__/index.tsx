/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import * as React from 'react';

import { Props } from '..';

export class GitConfigSectionUser extends React.PureComponent<Props> {
  render(): React.ReactElement {
    const { config, onChange } = this.props;
    return (
      <div>
        <div data-testid="user-email">{config.user.email}</div>
        <div data-testid="user-name">{config.user.name}</div>
        <button
          onClick={() => onChange({ user: { email: 'new-user@che', name: 'new user' } }, true)}
        >
          Change Email Valid
        </button>
      </div>
    );
  }
}
