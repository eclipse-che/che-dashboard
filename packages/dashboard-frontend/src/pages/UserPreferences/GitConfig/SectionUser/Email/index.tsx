/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { FormGroup, TextInput, ValidatedOptions } from '@patternfly/react-core';
import { ExclamationCircleIcon } from '@patternfly/react-icons';
import React from 'react';

import { InputGroupExtended } from '@/components/InputGroupExtended';

const ERROR_REQUIRED_VALUE = 'A value is required.';
const MAX_LENGTH = 128;
const ERROR_MAX_LENGTH = `The value is too long. The maximum length is ${MAX_LENGTH} characters.`;
const REGEX = /^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}$/;
const ERROR_INVALID_EMAIL = 'The value is not a valid email address.';

export type Props = {
  value: string;
  onChange: (value: string) => void;
};
export type State = {
  errorMessage?: string;
  value: string;
  validated: ValidatedOptions | undefined;
};

export class GitConfigUserEmail extends React.PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      value: props.value,
      validated: undefined,
    };
  }

  private handleChange(value: string): void {
    this.validate(value);
    this.setState({ value });
  }

  private handleSave(): void {
    const { value } = this.state;
    this.props.onChange(value);
  }

  private handleCancel(): void {
    const { value } = this.props;
    this.setState({
      value,
      validated: undefined,
    });
  }

  private validate(value: string): void {
    if (value.length === 0) {
      this.setState({
        errorMessage: ERROR_REQUIRED_VALUE,
        validated: ValidatedOptions.error,
      });
      return;
    }
    if (value.length > MAX_LENGTH) {
      this.setState({
        errorMessage: ERROR_MAX_LENGTH,
        validated: ValidatedOptions.error,
      });
      return;
    }
    if (!REGEX.test(value)) {
      this.setState({
        errorMessage: ERROR_INVALID_EMAIL,
        validated: ValidatedOptions.error,
      });
      return;
    }
    this.setState({
      errorMessage: undefined,
      validated: ValidatedOptions.success,
    });
  }

  public render(): React.ReactElement {
    const { errorMessage, value, validated } = this.state;

    const fieldId = 'gitconfig-user-email';

    return (
      <FormGroup
        label="email"
        fieldId={fieldId}
        isRequired
        helperTextInvalid={errorMessage}
        helperTextIcon={<ExclamationCircleIcon />}
      >
        <InputGroupExtended
          readonly={false}
          value={value}
          validated={validated}
          onSave={() => this.handleSave()}
          onCancel={() => this.handleCancel()}
        >
          <TextInput
            id={fieldId}
            value={value}
            validated={validated}
            onChange={value => this.handleChange(value)}
          />
        </InputGroupExtended>
      </FormGroup>
    );
  }
}
