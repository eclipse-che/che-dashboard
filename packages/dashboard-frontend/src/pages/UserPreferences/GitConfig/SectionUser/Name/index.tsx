/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { FormGroup, TextInput, ValidatedOptions } from '@patternfly/react-core';
import { ExclamationCircleIcon } from '@patternfly/react-icons';
import React from 'react';

import { InputGroupExtended } from '@/components/InputGroupExtended';

const ERROR_REQUIRED_VALUE = 'A value is required.';
const MAX_LENGTH = 128;
const ERROR_MAX_LENGTH = `The value is too long. The maximum length is ${MAX_LENGTH} characters.`;

export type Props = {
  value: string;
  onChange: (value: string) => void;
};
export type State = {
  errorMessage?: string;
  value: string;
  validated: ValidatedOptions | undefined;
};

export class GitConfigUserName extends React.PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      value: props.value,
      validated: undefined,
    };
  }

  private handleChange(value: string): void {
    this.validate(value);
    this.setState({ value });
  }

  private handleSave(): void {
    const { value } = this.state;
    this.props.onChange(value);
  }

  private handleCancel(): void {
    const { value } = this.props;
    this.setState({ value });
  }

  private validate(value: string): void {
    if (value.length === 0) {
      this.setState({
        errorMessage: ERROR_REQUIRED_VALUE,
        validated: ValidatedOptions.error,
      });
      return;
    }
    if (value.length > MAX_LENGTH) {
      this.setState({
        errorMessage: ERROR_MAX_LENGTH,
        validated: ValidatedOptions.error,
      });
      return;
    }
    this.setState({
      errorMessage: undefined,
      validated: ValidatedOptions.success,
    });
  }

  public render(): React.ReactElement {
    const { errorMessage, value, validated } = this.state;

    const fieldId = 'gitconfig-user-name';

    return (
      <FormGroup
        label="name"
        fieldId={fieldId}
        isRequired
        helperTextInvalid={errorMessage}
        helperTextIcon={<ExclamationCircleIcon />}
      >
        <InputGroupExtended
          readonly={false}
          value={value}
          validated={validated}
          onSave={() => this.handleSave()}
          onCancel={() => this.handleCancel()}
        >
          <TextInput
            id={fieldId}
            value={value}
            validated={validated}
            onChange={value => this.handleChange(value)}
            onSubmit={() => this.handleSave()}
          />
        </InputGroupExtended>
      </FormGroup>
    );
  }
}
