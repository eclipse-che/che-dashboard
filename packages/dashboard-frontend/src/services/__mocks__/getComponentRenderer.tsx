/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

export default function <T extends Array<unknown>>(
  getComponent: (..._args: T) => React.ReactElement,
): {
  createSnapshot: (...args: T) => renderer.ReactTestRenderer;
  renderComponent: (...args: T) => {
    reRenderComponent: (..._args: T) => void;
  };
} {
  return {
    createSnapshot: (...args: T) => {
      return renderer.create(getComponent(...args));
    },
    renderComponent: (...args: T) => {
      const res = render(getComponent(...args));
      return {
        reRenderComponent: (..._args: T) => {
          res.rerender(getComponent(..._args));
        },
      };
    },
  };
}
