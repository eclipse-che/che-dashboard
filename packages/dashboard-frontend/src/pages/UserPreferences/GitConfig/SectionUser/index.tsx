/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { Form, FormSection, Panel, PanelMain, PanelMainBody } from '@patternfly/react-core';
import * as React from 'react';

import { GitConfigUserEmail } from '@/pages/UserPreferences/GitConfig/SectionUser/Email';
import { GitConfigUserName } from '@/pages/UserPreferences/GitConfig/SectionUser/Name';
import { GitConfigUser } from '@/store/GitConfig';

export type Props = {
  config: GitConfigUser;
  onChange: (gitConfigUser: GitConfigUser) => void;
};

export class GitConfigSectionUser extends React.PureComponent<Props> {
  private handleChange(partialConfig: Partial<GitConfigUser>): void {
    const { config, onChange } = this.props;

    onChange({
      ...config,
      ...partialConfig,
    });
  }

  public render(): React.ReactElement {
    const { config } = this.props;
    return (
      <Panel>
        <PanelMain>
          <PanelMainBody>
            <Form isHorizontal onSubmit={e => e.preventDefault()}>
              <FormSection title="[user]" label="user">
                <GitConfigUserName
                  value={config.name}
                  onChange={name => this.handleChange({ name })}
                />
                <GitConfigUserEmail
                  value={config.email}
                  onChange={email => this.handleChange({ email })}
                />
              </FormSection>
            </Form>
          </PanelMainBody>
        </PanelMain>
      </Panel>
    );
  }
}
