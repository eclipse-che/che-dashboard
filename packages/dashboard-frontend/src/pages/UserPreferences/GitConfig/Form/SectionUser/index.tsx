/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { FormSection, Panel, PanelMain, PanelMainBody } from '@patternfly/react-core';
import * as React from 'react';

import { GitConfigUserEmail } from '@/pages/UserPreferences/GitConfig/Form/SectionUser/Email';
import { GitConfigUserName } from '@/pages/UserPreferences/GitConfig/Form/SectionUser/Name';
import { GitConfig } from '@/store/GitConfig';

export type Props = {
  config: GitConfig;
  isLoading: boolean;
  onChange: (gitConfig: GitConfig, isValid: boolean) => void;
};

export class GitConfigSectionUser extends React.PureComponent<Props> {
  private handleChange(partialConfigUser: Partial<GitConfig['user']>, isValid: boolean): void {
    const { config, onChange } = this.props;

    onChange(
      {
        ...config,
        user: {
          ...config.user,
          ...partialConfigUser,
        },
      },
      isValid,
    );
  }

  public render(): React.ReactElement {
    const { config, isLoading } = this.props;
    return (
      <Panel>
        <PanelMain>
          <PanelMainBody>
            <FormSection title="[user]" label="user">
              <GitConfigUserName
                isLoading={isLoading}
                value={config.user.name}
                onChange={(name, isValid) => this.handleChange({ name }, isValid)}
              />
              <GitConfigUserEmail
                isLoading={isLoading}
                value={config.user.email}
                onChange={(email, isValid) => this.handleChange({ email }, isValid)}
              />
            </FormSection>
          </PanelMainBody>
        </PanelMain>
      </Panel>
    );
  }
}
