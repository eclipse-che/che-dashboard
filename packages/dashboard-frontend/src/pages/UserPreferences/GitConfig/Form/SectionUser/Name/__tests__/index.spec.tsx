/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { ValidatedOptions } from '@patternfly/react-core';
import { StateMock } from '@react-mock/state';
import userEvent from '@testing-library/user-event';
import * as React from 'react';

import { GitConfigUserName, State } from '@/pages/UserPreferences/GitConfig/Form/SectionUser/Name';
import getComponentRenderer, { screen, waitFor } from '@/services/__mocks__/getComponentRenderer';

jest.mock('@/components/InputGroupExtended');

const { createSnapshot, renderComponent } = getComponentRenderer(getComponent);

const mockOnChange = jest.fn();

describe('GitConfigUserName', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('snapshot, not loading', () => {
    const snapshot = createSnapshot('user one', false);
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  test('snapshot, loading', () => {
    const snapshot = createSnapshot('user one', true);
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  it('should fail validation if value is empty', async () => {
    renderComponent('user one', false);

    const textInput = screen.getByRole('textbox');
    await userEvent.clear(textInput);

    expect(screen.getByTestId('validated')).toHaveTextContent(ValidatedOptions.error);
  });

  it('should fail validation if value is too long', async () => {
    renderComponent('user one', false);

    const textInput = screen.getByRole('textbox');
    await userEvent.click(textInput);
    await userEvent.paste('a'.repeat(129));

    expect(screen.getByTestId('validated')).toHaveTextContent(ValidatedOptions.error);
  });

  it('should reset validation', async () => {
    const initialValue = 'user name';
    const localState: Partial<State> = {
      value: '',
      validated: ValidatedOptions.error,
    };

    const { reRenderComponent } = renderComponent(initialValue, true, localState);

    reRenderComponent(initialValue, false, localState);

    await waitFor(() =>
      expect(screen.getByTestId('validated')).toHaveTextContent(ValidatedOptions.default),
    );
  });

  it('should handle value changing', async () => {
    renderComponent('user one', false);

    const textInput = screen.getByRole('textbox');
    await userEvent.click(textInput);
    await userEvent.paste(' two');

    expect(mockOnChange).toHaveBeenCalledWith('user one two', true);
  });
});

function getComponent(
  value: string,
  isLoading: boolean,
  localState?: Partial<State>,
): React.ReactElement {
  if (localState) {
    return (
      <StateMock state={localState}>
        <GitConfigUserName isLoading={isLoading} value={value} onChange={mockOnChange} />
      </StateMock>
    );
  }
  return <GitConfigUserName isLoading={isLoading} value={value} onChange={mockOnChange} />;
}
