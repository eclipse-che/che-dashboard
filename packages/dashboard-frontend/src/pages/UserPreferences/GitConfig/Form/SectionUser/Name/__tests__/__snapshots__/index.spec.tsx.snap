// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitConfigUserName snapshot, loading 1`] = `
<div
  className="pf-c-form__group"
>
  <div
    className="pf-c-form__group-label"
  >
    <label
      className="pf-c-form__label"
      htmlFor="gitconfig-user-name"
    >
      <span
        className="pf-c-form__label-text"
      >
        name
      </span>
      <span
        aria-hidden="true"
        className="pf-c-form__label-required"
      >
         
        *
      </span>
    </label>
     
  </div>
  <div
    className="pf-c-form__group-control"
  >
    <div>
      <input
        aria-invalid={false}
        aria-label={null}
        className="pf-c-form-control"
        data-ouia-component-id="OUIA-Generated-TextInputBase-2"
        data-ouia-component-type="PF4/TextInput"
        data-ouia-safe={true}
        disabled={true}
        id="gitconfig-user-name"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value="user one"
      />
      <span
        data-testid="validated"
      >
        default
      </span>
      <button
        aria-disabled={false}
        aria-label={null}
        className="pf-c-button pf-m-primary"
        data-ouia-component-id="OUIA-Generated-Button-primary-2"
        data-ouia-component-type="PF4/Button"
        data-ouia-safe={true}
        data-testid="button-remove"
        disabled={false}
        role={null}
        type="button"
      />
    </div>
  </div>
</div>
`;

exports[`GitConfigUserName snapshot, not loading 1`] = `
<div
  className="pf-c-form__group"
>
  <div
    className="pf-c-form__group-label"
  >
    <label
      className="pf-c-form__label"
      htmlFor="gitconfig-user-name"
    >
      <span
        className="pf-c-form__label-text"
      >
        name
      </span>
      <span
        aria-hidden="true"
        className="pf-c-form__label-required"
      >
         
        *
      </span>
    </label>
     
  </div>
  <div
    className="pf-c-form__group-control"
  >
    <div>
      <input
        aria-invalid={false}
        aria-label={null}
        className="pf-c-form-control"
        data-ouia-component-id="OUIA-Generated-TextInputBase-1"
        data-ouia-component-type="PF4/TextInput"
        data-ouia-safe={true}
        disabled={false}
        id="gitconfig-user-name"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value="user one"
      />
      <span
        data-testid="validated"
      >
        default
      </span>
      <button
        aria-disabled={false}
        aria-label={null}
        className="pf-c-button pf-m-primary"
        data-ouia-component-id="OUIA-Generated-Button-primary-1"
        data-ouia-component-type="PF4/Button"
        data-ouia-safe={true}
        data-testid="button-remove"
        disabled={false}
        role={null}
        type="button"
      />
    </div>
  </div>
</div>
`;
